stubs/redis/redis/client.pyi

```diff
@@ -285,6 +287,7 @@ class Redis(AbstractRedis, RedisModuleCommands, CoreCommands[_StrType], Sentinel
         name: _Key,
         timeout: float | None,
         sleep: float,
+        blocking: bool,
```

Comment:
That's invalid syntax. An argument without a default value can't be after an argument that has a default value, i.e. all non-defaulted arguments must be in the beginning.

The usual workaround is to split this into two overloads: you either pass `lock_class` positionally, so `__lock_class` and no defaults for arguments before it, or you pass it by keyword so `*` before `lock_class`.

Reply:
Just realized that you don't *have* to use `__lock_class` in the "passed as a positional argument" case. Then it will also match some calls where it's passed as a keyword, but that's fine because the return types are the same.

Reply:
Oh of course, thanks, totally blanked on that one